### DISCUSSION TITLE
iModel Download Progress Callback

### DISCUSSION THREAD

**Discussion by nn15296** [iTwin/itwinjs-core#733] [Q&A] [answered]
The latest version of imodeljs no longer seems to support progress callbacks to monitor download progress when downloading briefcases - as was introduced in imodel version 2.0.0. Is there support for this feature in any other capacity?

**Comment by kabentley** [comment, marked as answer]
what api are you using to download a briefcase?

`BriefcaseManager.downloadBriefcase` has an `onProgress` function that is called during downloading, but it's backend only.

The `NativeApp.requestDownloadBriefcase` frontend method has a `progress` argument.

**Reply by nn15296** [reply]
I'm using the Briefcase Manager (https://www.itwinjs.org/reference/imodeljs-backend/imodels/briefcasemanager/downloadbriefcasestatic/) but can't find the onProgress function.

**Reply by kabentley** [reply]
There must be something wrong with the documentation generator, it doesn't give the definition of `RequestNewBriefcaseArg`.

Look in `BriefcaseManager.ts` and you'll see the definition of its member `onProgress`

**Reply by kabentley** [reply]
@williamkbentley
